src/filter.ts

```diff
@@ -58,10 +58,17 @@ export class Filter {
     for (const [key, patterns] of Object.entries(this.rules)) {
       result[key] = files.filter(file => this.isMatch(file, patterns))
     }
+
+    if (!this.rules.hasOwnProperty('other')) {
+      const matchingFilenamesList = Object.values(result).flatMap(filteredFiles => filteredFiles.map(file => file.filename))
+      const matchingFilenamesSet = new Set(matchingFilenamesList)
+      result.other = files.filter(file => !matchingFilenamesSet.has(file.filename))
+    }
```

Comment:
If there already was a filter group named `other`, it won't be overwritten. If the config did not have it, adding a new non-declared group should be safe - the keys are not in JSON, so it is unlikely that there exists logic in a step that uses paths-filter output that discovers the keys rather than uses the hardcoded names.